src/classes/TriggerHandler.cls

```diff
@@ -67,6 +71,20 @@ public abstract class TriggerHandler {
         if(!this.isTriggerExecuting || this.context == null) throw new Exceptions.TriggerHandlerException(errorMessage);
     }
 
+    private Boolean shouldExecuteTriggers() {
+        TriggerSettings__c triggerSettings = TriggerSettings__c.getInstance();
+
+        if(triggerSettings.Id == null) {
+            // If there's no ID, then there are settings setup for the current user at the user, profile or org level
+            // Upsert the org defaults - the default field values will be used
+            upsert TriggerSettings__c.getOrgDefaults();
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add one to ensure exceptions are handled properly.